lib/dynamoid/adapter_plugin/aws_sdk_v3.rb

```diff
@@ -92,6 +88,21 @@ def connection_config
         @connection_hash
       end
 
+      def add_credentials_to_hash(connection_hash)
+        # if credentials are passed, they already contain access key & secret key
+        if Dynamoid::Config.credentials?
+          connection_hash[:credentials] = Dynamoid::Config.credentials
+        else
+          # otherwise, pass access key & secret key for credentials creation
+          if Dynamoid::Config.access_key?
+            connection_hash[:access_key_id] = Dynamoid::Config.access_key
+          end
+          if Dynamoid::Config.secret_key?
+            connection_hash[:secret_access_key] = Dynamoid::Config.secret_key
+          end
+        end
+      end
```

Comment:
Made a new method - tooling complained about complexity. Open to other ideas!

Reply:
Ah, moving it out didn't reduce the method complexity. I guess this alone has a complexity of 6. Let me know how I can remedy this.